go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -1141,12 +1188,12 @@ func FilterIgnoredTables(ctx context.Context, tables []TableName, roots Roots) (
 
 // ValidateTagUniqueness checks for tag collisions between the given table and the set of tables in then given root.
 func ValidateTagUniqueness(ctx context.Context, root RootValue, tableName string, table *Table) error {
-	prev, ok, err := root.GetTable(ctx, TableName{Name: tableName})
+	prevTName := TableName{Name: tableName}
+	prevHash, err := root.GetTableSchemaHash(ctx, prevTName, table.overriddenSchema != nil)
```

Comment:
This logic is incorrect now, or at least subject to lots of assumptions. In the case that the cache is empty when this is called, you'll skip the checks below even when the schema has changed. Need to find a different way to do this.

Reply:
I'm confused, the interface `GetTableSchemaHash` isn't checking a cache, it returns the accurate schema hash for a table if it exists. If the value was previously computed for the root value, it avoids re-deserializing. The `override` parameter was explicit in the first version because something about `overridden=true` that makes root values mutable related specifically to the schema parameter.